src/app/ReadClient.cpp

```diff
@@ -617,6 +617,7 @@ CHIP_ERROR ReadClient::ProcessReportData(System::PacketBufferHandle && aPayload,
             // a subscription and have a valid value for mMaxInterval which the function
             // relies on.
             //
+            mpCallback.RefreshLiveness(*this);
```

Comment:
Is the goal of this callback to kick a re subscription when in the `InactiveICDSubscription` state? If so then this will not get called because of the `IsSubscriptionActive()` check on line 613.

Reply:
No

One of the goal is to kick a resubscription on possibe **other** subscriptions in `InactiveICDSubscription` state.

Since we just received a report message, there is no need to kick this subscription, but other subscriptions might be out of sync.